cpp/src/arrow/adapters/orc/adapter_util.cc

```diff
@@ -315,13 +344,662 @@ Status AppendBatch(const liborc::Type* type, liborc::ColumnVectorBatch* batch,
+    case arrow::Type::type::LARGE_LIST:
+      return WriteListBatch<arrow::LargeListArray>(array, orc_offset,
+                                                   column_vector_batch);
+    case arrow::Type::type::FIXED_SIZE_LIST:
+      return WriteListBatch<arrow::FixedSizeListArray>(array, orc_offset,
+                                                       column_vector_batch);
+    case arrow::Type::type::MAP:
+      return WriteMapBatch(array, orc_offset, column_vector_batch);
+    default: {
+      return arrow::Status::Invalid("Unknown or unsupported Arrow type: ",
+                                    array.type()->ToString());
+    }
+  }
+  return arrow::Status::OK();
+}
+
+arrow::Result<ORC_UNIQUE_PTR<liborc::Type>> GetORCType(const arrow::DataType& type) {
+  arrow::Type::type kind = type.id();
+  switch (kind) {
+    case arrow::Type::type::BOOL:
+      return liborc::createPrimitiveType(liborc::TypeKind::BOOLEAN);
+    case arrow::Type::type::INT8:
+      return liborc::createPrimitiveType(liborc::TypeKind::BYTE);
+    case arrow::Type::type::INT16:
+      return liborc::createPrimitiveType(liborc::TypeKind::SHORT);
+    case arrow::Type::type::INT32:
+      return liborc::createPrimitiveType(liborc::TypeKind::INT);
+    case arrow::Type::type::INT64:
+      return liborc::createPrimitiveType(liborc::TypeKind::LONG);
+    case arrow::Type::type::FLOAT:
+      return liborc::createPrimitiveType(liborc::TypeKind::FLOAT);
+    case arrow::Type::type::DOUBLE:
+      return liborc::createPrimitiveType(liborc::TypeKind::DOUBLE);
+    // Use STRING instead of VARCHAR for now, both use UTF-8
+    case arrow::Type::type::STRING:
+    case arrow::Type::type::LARGE_STRING:
+      return liborc::createPrimitiveType(liborc::TypeKind::STRING);
+    case arrow::Type::type::BINARY:
+    case arrow::Type::type::LARGE_BINARY:
+    case arrow::Type::type::FIXED_SIZE_BINARY:
+      return liborc::createPrimitiveType(liborc::TypeKind::BINARY);
+    case arrow::Type::type::DATE32:
+      return liborc::createPrimitiveType(liborc::TypeKind::DATE);
+    case arrow::Type::type::DATE64:
+    case arrow::Type::type::TIMESTAMP:
+      return liborc::createPrimitiveType(liborc::TypeKind::TIMESTAMP);
+    case arrow::Type::type::DECIMAL128: {
+      const uint64_t precision = static_cast<uint64_t>(
+          static_cast<const arrow::Decimal128Type&>(type).precision());
+      const uint64_t scale =
+          static_cast<uint64_t>(static_cast<const arrow::Decimal128Type&>(type).scale());
+      return liborc::createDecimalType(precision, scale);
+    }
+    case arrow::Type::type::LIST:
+    case arrow::Type::type::FIXED_SIZE_LIST:
+    case arrow::Type::type::LARGE_LIST: {
+      std::shared_ptr<arrow::DataType> arrow_child_type =
+          static_cast<const arrow::BaseListType&>(type).value_type();
+      ORC_UNIQUE_PTR<liborc::Type> orc_subtype =
+          GetORCType(*arrow_child_type).ValueOrDie();
```

Comment:
Please don't use `ValueOrDie` unless errors should never happen in practice. Instead:
```c++
ARROW_ASSIGN_OR_RAISE(auto orc_child_type, GetOrcType(*arrow_child_type));
```

Reply:
Really thanks!

Reply:
All instances of `ValueOrDie` has been replaced by `ARROW_ASSIGN_OR_RAISE` or `EXPECT_OK_AND_ASSIGN` for regular code and test code respectively throughout my code base.